dotnet/samples/Concepts/Search/VectorStore_TextSearch.cs

```diff
@@ -0,0 +1,182 @@
+            "Semantic Kernel is a new AI SDK, and a simple and yet powerful programming model that lets you add large language capabilities to your app in just a matter of minutes. It uses natural language prompting to create and execute semantic kernel AI tasks across multiple languages and platforms.",
+            "In this guide, you learned how to quickly get started with Semantic Kernel by building a simple AI agent that can interact with an AI service and run your code. To see more examples and learn how to build more complex AI agents, check out our in-depth samples."
+        ];
+        var vectorizedSearch = await CreateCollectionFromListAsync<Guid, DataModel>(
+            vectorStore, collectionName, lines, textEmbeddingGeneration, CreateRecord);
+
+        // Create a text search instance using the volatile vector store.
+        var stringMapper = new DataModelTextSearchStringMapper();
+        var resultMapper = new DataModelTextSearchResultMapper();
+        var textSearch = new VectorStoreRecordTextSearch<DataModel>(vectorizedSearch, textEmbeddingGeneration, stringMapper, resultMapper);
+
+        var query = "What is the Semantic Kernel?";
+
+        // Search and return results as a string items
+        KernelSearchResults<string> stringResults = await textSearch.SearchAsync(query, new() { Count = 2, Offset = 0 });
+        Console.WriteLine("--- String Results ---\n");
+        await foreach (string result in stringResults.Results)
+        {
+            Console.WriteLine(result);
+            WriteHorizontalRule();
+        }
+
+        // Search and return results as TextSearchResult items
+        KernelSearchResults<TextSearchResult> textResults = await textSearch.GetTextSearchResultsAsync(query, new() { Count = 2, Offset = 0 });
+        Console.WriteLine("\n--- Text Search Results ---\n");
+        await foreach (TextSearchResult result in textResults.Results)
+        {
+            Console.WriteLine($"Name:  {result.Name}");
+            Console.WriteLine($"Value: {result.Value}");
+            Console.WriteLine($"Link:  {result.Link}");
+            WriteHorizontalRule();
+        }
+
+        // Search and return s results as BingWebPage items
+        KernelSearchResults<object> fullResults = await textSearch.GetSearchResultsAsync(query, new() { Count = 2, Offset = 0 });
+        Console.WriteLine("\n--- DataModel Results ---\n");
+        await foreach (DataModel result in fullResults.Results)
+        {
+            Console.WriteLine($"Key:         {result.Key}");
+            Console.WriteLine($"Text:        {result.Text}");
+            Console.WriteLine($"Embedding:   {result.Embedding.Length}");
+            WriteHorizontalRule();
+        }
+    }
+
+    /// <summary>
+    /// String mapper which converts a DataModel to a string.
+    /// </summary>
+    private sealed class DataModelTextSearchStringMapper : ITextSearchStringMapper
```

Comment:
It would be nice to have an implementation of `ITextSearchStringMapper` where we can just pass a lambda to the constructor, so someone doesn't have to create their own class.

Reply:
Added to this task: https://github.com/microsoft/semantic-kernel/issues/6725